tests/strategies/test_exit.py

```diff
@@ -187,37 +187,72 @@ def test_exit_by_trailing_stop(market, signal, entries):
     pd.testing.assert_frame_equal(trades, expected)
 
 
-def test_exit_at_loss_and_gain(market, signal, entries):
+@pytest.mark.parametrize(
+    "gain_threshold,loss_threshold,expected_data",
+    [
+        (
+            100,
+            1,
+            [
+                [1.0],  # U
+                [-1.0],  # D
+                [1.0],  # Hit loss - Sell
+                [0.0],  # (Hit max holding) and U
+                [1.0],  # U
+                [-1.0],  # D
+                [-1.0],  # D
+                [0.0],  # and so on
+                [1.0],  #
+                [1.0],  #
+                [1.0],  #
+                [-2.0],  #
+                [-1],  #
+                [-1],  #
+                [1.0],  #
+                [1.0],  #
+                [1.0],  #
+                [1.0],  #
+                [-3.0],  #
+            ],
+        ),
+        (
+            1,
+            100,
+            [
+                [1.0],
+                [-2.0],
+                [1.0],
+                [1.0],
+                [-2.0],
+                [-1.0],
+                [1.0],
+                [2.0],
+                [0.0],
+                [0.0],
+                [-2.0],
+                [-1.0],
+                [-1.0],
+                [1.0],
+                [1.0],
+                [1.0],
+                [1.0],
+                [0.0],
+                [0.0],
+                [-1.0],
```

Comment:
Did you check that the logic is correct by unittest? In that case, you can merge it.

Reply:
I double checked and it seems that logic is followed by unittests. I'll meerge it